Prox/Prox/PlaceCarousel/PlaceCarouselViewController.swift

```diff
@@ -515,24 +450,25 @@ extension PlaceCarouselViewController: PlacesProviderDelegate {
     }
 
     func placesProvider(_ controller: PlacesProvider, didReceivePlaces places: [Place]) {
-        let wasEmpty = self.places.isEmpty
-        self.places = places
-
-        (self.presentedViewController as? PlaceDetailViewController)?.placesUpdated()
 
-        if wasEmpty && !places.isEmpty {
+        // TODO: how do we make sure the user wasn't interacting?
+        headerView.numberOfPlacesLabel.text = "\(places.count) place" + (places.count != 1 ? "s" : "")
+        placeCarousel.refresh()
 
-            // Wrap the openClosedPlace in an async block to make sure its queued after the
-            // carousel's refresh so the cells load before we invoke the transition
-            DispatchQueue.main.async {
-                self.openFirstPlace()
-            }
-        }
+        (self.presentedViewController as? PlaceDetailViewController)?.placesUpdated()
     }
 
     func placesProviderDidTimeout(_ controller: PlacesProvider) {
         showErrorMessageIfNoPlaces()
     }
+
+    func placesProviderDidFetchFirstPlace() {
+        // Wrap the openClosedPlace in an async block to make sure its queued after the
+        // carousel's refresh so the cells load before we invoke the transition
```

Comment:
I think the comment was originally added by @sleroux when he did the animations - I just copied it along with the relevant code when I moved it.